src/test/java/io/camunda/testing/assertions/IncidentAssertTest.java

```diff
@@ -0,0 +1,449 @@
+package io.camunda.testing.assertions;
+
+import static io.camunda.testing.assertions.BpmnAssert.assertThat;
+import static io.camunda.testing.util.Utilities.deployProcess;
+import static io.camunda.testing.util.Utilities.startProcessInstance;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.testing.extensions.ZeebeAssertions;
+import io.camunda.testing.util.Utilities.ProcessPackLoopingServiceTask;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.ActivateJobsResponse;
+import io.camunda.zeebe.client.api.response.ActivatedJob;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import io.camunda.zeebe.protocol.record.value.ErrorType;
+import java.util.Collections;
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.StringAssert;
+import org.camunda.community.eze.RecordStreamSource;
+import org.camunda.community.eze.ZeebeEngine;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@ZeebeAssertions
+class IncidentAssertTest {
+
+  public static final String WRONG_VALUE = "wrong value";
+  public static final String ERROR_CODE = "error";
+  public static final String ERROR_MESSAGE = "error occurred";
+
+  private ZeebeClient client;
+  private ZeebeEngine engine;
+
+  // These tests are for testing assertions as well as examples for users
+  @Nested
+  class HappyPathTests {
+
+    private RecordStreamSource recordStreamSource;
+
+    @Test
+    void testHasErrorType() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
```

Comment:
🔧 Instead of creating a variable we could chain the actual assertion. Idem for the other tests.

Reply:
I like it better this way, but not by much. So happy to discuss this if you disagree